src/build.proj

```diff
@@ -5,6 +5,7 @@
   <ItemGroup>
     <Project Condition="$(SkipSOS) != 'true'" Include="ToolBox\SOS\NETCore\SOS.NETCore.csproj" />
     <Project Include="System.Private.CoreLib\System.Private.CoreLib.csproj" />
+    <Project Condition="'$(DotNetBuildFromSource)' != 'true' AND '$(__BuildManagedTools)' == '1'" Include="tools/r2rdump/R2RDump.csproj" />
```

Comment:
In build.cmd, the `%__BuildSOS%` environment variable is used to set `$(SkipSOS)` to true. You could do the same thing here with the `%__BuildManagedTools%` variable so here you could test whether or not `$(BuildManagedTools)` is true.